src/userland/userland.c

```diff
@@ -17,15 +9,112 @@ uint64_t strlen(uint8_t* s) {
 	return len;
 }
 
-#define EVER ;;
+void sys_print(uint8_t* s, uint64_t len) {
+	// $0 is immediate value 0 in AT-T syntax.
+	asm volatile("mov $0, %%rax\n"
+	             "mov %0, %%rdi\n"
+	             "mov %1, %%rsi\n"
+	             "syscall\n" ::"r"(s),
+	             "r"(len)
+	             : "rax", "rdi", "rsi");
+}
+
+uint8_t sys_getch() {
+	uint64_t c;
+	asm volatile("mov $1, %%rax\n"
+	             "mov %0, %%rdi\n"
+	             "syscall\n" ::"r"(&c)
+	             : "rax", "rdi");
+	return c & 0xff;
+}
+
+void get_line(uint8_t line[128]) {
+	uint8_t  c;
+	uint64_t i = 0;
+	while ((c = sys_getch())) {
+		if ('\n' == c) {
+			break;
+		}
+		line[i] = c;
+		i++;
+	}
+}
+
+void debug_interrupt() {
+	// NOTE: This will actually throw a general protection fault, since it's a
+	// privileged instruction. So we just hook this empty void stub instead and
+	// pretend that it works!
+
+	/* asm volatile("int3"); */
+}
+
+#define EVER                                                                   \
```

Comment:
HAHAHA! den fick leva kvar. `for EVER`

Reply:
clang-format verkar gilla göra den utdragen också :p